js/core/src/registry.ts

```diff
@@ -126,10 +126,19 @@ type ActionsRecord = Record<string, Action<z.ZodTypeAny, z.ZodTypeAny>>;
  * Returns all actions in the registry.
  */
 export async function listActions(): Promise<ActionsRecord> {
+  await maybeInitializeAllPlugins();
+  return Object.assign({}, actionsById());
+}
+
+let initializedAllPlugins = false;
+export async function maybeInitializeAllPlugins() {
```

Comment:
This reads weird. It definitely *does* initialize all plugins, it's just that it may have already been done. I would suggest either `initializeAllPlugins` or `allPluginsInitialized`.

Reply:
renamed